kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesAttributesExtractor.java

```diff
@@ -181,10 +183,12 @@ public AttributeSet extract(HasMetadata hasMetadata) {
     }
     if (!Utils.isNullOrEmpty(hasMetadata.getMetadata().getName())) {
       metadataAttributes = metadataAttributes.add(new Attribute(NAME, hasMetadata.getMetadata().getName()));
+      metadataAttributes = metadataAttributes.add(new Attribute(METADATA_NAME, hasMetadata.getMetadata().getName()));
     }
 
     if (!Utils.isNullOrEmpty(hasMetadata.getMetadata().getNamespace())) {
       metadataAttributes = metadataAttributes.add(new Attribute(NAMESPACE, hasMetadata.getMetadata().getNamespace()));
+      metadataAttributes = metadataAttributes.add(new Attribute(METADATA_NAMESPACE, hasMetadata.getMetadata().getNamespace()));
```

Comment:
Both this and `METADATA_NAME` attributes (L56, L58, L186) seem redundant of the already present name and namespace attributes. I don't see them used anywhere, are they really needed?

Reply:
Yes, they are both needed to make the existing attributes names match the actual field names. 
When searching by name (`client.services().inNamespace("ns1").withName("svc1").get()`) it will try to match an attribute `name` with field `name` while if searching with field `client.services().inNamespace("ns1").withField("metadata.name", "svc1").list()`) it will try to match with `metadata.name`, so we need both attributes in the memory database to be able to find the resources by 2 ways.